documentation/DATA_FORMAT.md

```diff
@@ -0,0 +1,374 @@
+        <track number for id 1>,
+        <track number for id 2>,
+        ...
+    ]
+}
+```
+
+<details>
+<summary><b>[Show me an example!]</b></summary>
+
+---
+
+For example, if there were the following two tracks in some dataset, the track file might look something like this.
+
+| Track # | Object IDs |
+| ------- | ---------- |
+| 1       | 0, 1, 4    |
+| 2       | 2, 3, 5    |
+
+Note that the object IDs in a track are not guaranteed to be sequential!
+
+```
+--tracks.json--
+{
+    "data": [
+        1, // 0
+        1, // 1
+        2, // 2
+        2, // 3
+        1, // 4
+        2  // 5
+    ]
+}
+```
+
+---
+
+</details>
+
+### 2. Times
+
+The times JSON is similar to the tracks JSON. It also contains a `data` array that maps from object IDs to the frame number that they appear on.
+
+```
+--times.json--
+{
+    "data": [
+        <frame number for id 0>,
+        <frame number for id 1>,
+        <frame number for id 2>,
+        ...
+    ]
+}
+```
+
+### 3. Frames
+
+_Example frame:_
+![](./frame_example.png)
+_Each unique color in this frame is a different object ID._
+
+**Frames** are image textures that store the object IDs for each time step in the time series. Each pixel in the image can encode a single object ID in its RGB value (`object ID = R + G*256 + B*256*256 - 1`), and background pixels are `#000000` (black).
+
+Additional notes:
+
+- Encoded object ID's in the frame data start at `1` instead of `0`, because `#000000` (black) is reserved for the background.
+- The highest object ID that can currently be represented is `16,843,007`.
+  - If the **total number of segmented objects** for an entire time series exceeds this number, it is possible to remove this limit. [Submit an issue](https://github.com/allen-cell-animated/nucmorph-colorizer/issues) or send us a message!
+
+There should be one frame for every time step in the time series, and they must all be listed in order in the **manifest** file to be included in the dataset.
+
+<details>
+<summary><b>[Show me an example!]</b></summary>
+
+---
+
+Let's say we have a simple 3x3 image, and the center pixel is mapped to the object ID `640` surrounded by the background.
+
+The calculation for the RGB value would follow this process.
+
+1. Add one to the object ID, because of 1-based indexing. (`ID = 641`)
+2. Get the Red channel value. (`R = ID % 256 = 641 % 256 = 129`)
+3. Get the Green channel value. (`G = ⌊ID / 256⌋ % 256 = 1 % 256 = 2`)
+4. Get the Blue channel value. (`B = ⌊ID / (256^2)⌋ = ⌊641 / (256^2)⌋ = 0`)
+
+The RGB value for ID `640` will be `RGB(129, 2, 0)`, or `#810200`.
+
+The resulting frame would look like this:
+
+![](./frame_example_simple.png)
+
+---
+
+</details>
+
+### 4. Features
+
+Datasets can contain any number of `features`, which are a numeric value assigned to each object ID in the dataset. Each feature file corresponds to a single column of data.
+
+Features must also provide a `min` and `max` range property.
```

Comment:
```suggestion
Features should also provide a `min` and `max` range property.
```
Technically, the app will derive `min` and `max` as needed, it'll just have to iterate over the whole array to do so